src/main/java/seedu/address/logic/EditUtil.java

```diff
@@ -0,0 +1,267 @@
+
+        public Optional<Email> getEmail() {
+            return Optional.ofNullable(email);
+        }
+
+        public void setAddress(Address address) {
+            this.address = address;
+        }
+
+        public Optional<Address> getAddress() {
+            return Optional.ofNullable(address);
+        }
+
+        public void setSchedule(Schedule schedule) {
+            this.schedule = schedule;
+        }
+
+        public Optional<Schedule> getSchedule() {
+            return Optional.ofNullable(schedule);
+        }
+
+        /**
+         * Sets {@code tags} to this object's {@code tags}.
+         * A defensive copy of {@code tags} is used internally.
+         */
+        public void setTags(Set<Tag> tags) {
+            this.tags = (tags != null) ? new HashSet<>(tags) : null;
+        }
+
+        /**
+         * Returns an unmodifiable tag set, which throws {@code UnsupportedOperationException}
+         * if modification is attempted.
+         * Returns {@code Optional#empty()} if {@code tags} is null.
+         */
+        public Optional<Set<Tag>> getTags() {
+            return (tags != null) ? Optional.of(Collections.unmodifiableSet(tags)) : Optional.empty();
+        }
+
+        @Override
+        public boolean equals(Object other) {
+            // short circuit if same object
+            if (other == this) {
+                return true;
+            }
+
+            // instanceof handles nulls
+            if (!(other instanceof EditPersonDescriptor)) {
+                return false;
+            }
+
+            // state check
+            EditPersonDescriptor e = (EditPersonDescriptor) other;
+
+            return getName().equals(e.getName())
+                    && getPhone().equals(e.getPhone())
+                    && getEmail().equals(e.getEmail())
+                    && getAddress().equals(e.getAddress())
+                    && getSchedule().equals(e.getSchedule())
+                    && getTags().equals(e.getTags());
+        }
+    }
+
+    public static class EditScheduleDescriptor {
+        private List<Event> events;
+
+        public EditScheduleDescriptor() {}
+
+        /**
+         * Copy constructor.
+         * A defensive copy of {@code tags} is used internally.
+         */
+        public EditScheduleDescriptor(EditScheduleDescriptor toCopy) {
+            setEvents(toCopy.events);
+        }
+
+        /**
+         * Sets {@code events} to this object's {@code events}.
+         * A defensive copy of {@code events} is used internally.
+         */
+        public void setEvents(List<Event> events) {
+            this.events = (events != null) ? new ArrayList<>(events) : null;
+        }
+
+        /**
+         * Returns an unmodifiable events list, which throws {@code UnsupportedOperationException}
+         * if modification is attempted.
+         * Returns {@code Optional#empty()} if {@code events} is null.
+         */
+        public Optional<List<Event>> getEvents() {
```

Comment:
I do not think that this should be returning Optional as events are the only field that we edit here. It would not be optional to have a List of events in this class, unlike EditPersonDescriptor or EditEventDescriptor